### PR TITLE
Use binary release of dep

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [ ] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [ ] I've tested my changes, adding unit tests where applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [ ] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [ ] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/1006

File: .travis.yml [hunk 1]
```diff
@@ -24,9 +26,11 @@ matrix:
     - env: MODE=breaking
     - go: master
 
+before_install:
+  - curl -L -o $GOPATH/bin/dep https://github.com/golang/dep/releases/download/v$DEP_VERSION/dep-linux-amd64 && chmod +x $GOPATH/bin/dep
```
Comment (vladbarosan) [inline .travis.yml]: we should add here a check on the SHA of the binary to make sure it is what we expect . for 0.4.1
it is  `31144e465e52ffbc0035248a10ddea61a09bf28b00784fd3fdd9882c8cbb2315` 

somethign like  :
```
- curl -fL -o $GOPATH/bin/dep https://github.com/golang/dep/releases/download/v$DEP_VERSION/dep-linux-amd64  && echo "$DEP_SHA $GOPATH/bin/dep" | sha256sum -c - && chmod +x $GOPATH/bin/dep
```

Comment (vladbarosan) [reply .travis.yml]: https://github.com/golang/dep/releases/download/v0.4.1/dep-linux-amd64.sha256

Comment (jhendrixMSFT) [reply .travis.yml]: This will verify the hash?

File: .travis.yml [hunk 2]
```diff
@@ -24,9 +26,11 @@ matrix:
     - env: MODE=breaking
     - go: master
 
+before_install:
+  - curl -fL -o $GOPATH/bin/dep https://github.com/golang/dep/releases/download/v$DEP_VERSION/dep-linux-amd64  && echo "$DEP_SHA $GOPATH/bin/dep" | sha256sum -c - && chmod +x $GOPATH/bin/dep
```
Comment (vladbarosan) [inline .travis.yml]: we need to define the $DEP_SHA value also

**Pull request merged**